src/snowflake/snowpark/modin/plugin/extensions/dataframe_overrides.py

```diff
@@ -7,23 +7,1474 @@
+from pandas._libs.lib import NoDefault, no_default
+from pandas._typing import (
+    AggFuncType,
+    AnyArrayLike,
+    Axes,
+    Axis,
+    CompressionOptions,
+    FilePath,
+    FillnaOptions,
+    IgnoreRaise,
+    IndexLabel,
+    Level,
+    PythonFuncType,
+    Renamer,
+    Scalar,
+    StorageOptions,
+    Suffixes,
+    WriteBuffer,
+)
+from pandas.core.common import apply_if_callable, is_bool_indexer
+from pandas.core.dtypes.common import (
+    infer_dtype_from_object,
+    is_bool_dtype,
+    is_dict_like,
+    is_list_like,
+    is_numeric_dtype,
+)
+from pandas.core.dtypes.inference import is_hashable, is_integer
+from pandas.core.indexes.frozen import FrozenList
+from pandas.io.formats.printing import pprint_thing
+from pandas.util._validators import validate_bool_kwarg
+
+from snowflake.snowpark.modin.pandas.api.extensions import register_dataframe_accessor
+from snowflake.snowpark.modin.pandas.groupby import (
+    DataFrameGroupBy,
+    validate_groupby_args,
+)
+from snowflake.snowpark.modin.pandas.snow_partition_iterator import (
+    SnowparkPandasRowPartitionIterator,
+)
+from snowflake.snowpark.modin.pandas.utils import (
+    create_empty_native_pandas_frame,
+    from_non_pandas,
+    from_pandas,
+    is_scalar,
+    raise_if_native_pandas_objects,
+    replace_external_data_keys_with_empty_pandas_series,
+    replace_external_data_keys_with_query_compiler,
+)
+from snowflake.snowpark.modin.plugin._internal.aggregation_utils import (
+    is_snowflake_agg_func,
+)
+from snowflake.snowpark.modin.plugin._internal.telemetry import (
+    snowpark_pandas_telemetry_method_decorator,
+    try_add_telemetry_to_attribute,
+)
+from snowflake.snowpark.modin.plugin._internal.utils import is_repr_truncated
+from snowflake.snowpark.modin.plugin._typing import ListLike
+from snowflake.snowpark.modin.plugin.utils.error_message import (
+    ErrorMessage,
+    dataframe_not_implemented,
+)
+from snowflake.snowpark.modin.plugin.utils.frontend_constants import (
+    DF_ITERROWS_ITERTUPLES_WARNING_MESSAGE,
+    DF_SETITEM_LIST_LIKE_KEY_AND_RANGE_LIKE_VALUE,
+    DF_SETITEM_SLICE_AS_SCALAR_VALUE,
+)
+from snowflake.snowpark.modin.plugin.utils.warning_message import WarningMessage
+from snowflake.snowpark.modin.utils import (
+    _inherit_docstrings,
+    hashable,
+    validate_int_kwarg,
+)
+from snowflake.snowpark.udf import UserDefinedFunction
+
+
+def register_dataframe_not_implemented():
+    def decorator(base_method: Any):
+        func = snowpark_pandas_telemetry_method_decorator(
+            dataframe_not_implemented()(base_method)
+        )
+        register_dataframe_accessor(base_method.__name__)(func)
+        return func
+
+    return decorator
+
+
+# === UNIMPLEMENTED METHODS ===
+# The following methods are not implemented in Snowpark pandas, and must be overridden on the
+# frontend. These methods fall into a few categories:
+# 1. Would work in Snowpark pandas, but we have not tested it.
```

Comment:
can we simply say all methods that we do not claim in supported API docs should be here?

Reply:
Eventually I think we should push the NotImplementedError down to the query compiler wherever possible rather than keeping the frontend method stub defined here. Keeping the stubs here just makes the migration easier for now.

Reply:
i think the answer to this is yes tat this moment. We are considering in the near future might potentially push this hanlization down to query compiler. 
However, @sfc-gh-joshi i am still questioning about this, because api not implemented is for the whole API, but modin have some checks performed at frontend layer, if we move it to query compiler layer, it just means the concrete implementation is missing, but user may still see some error if they didn't pass the correct parameter values inside etc. This behavior could be wired to uers.

Reply:
That's true, and I'm not sure if upstream Modin would be OK with pushing parameter validation to the query compiler in all cases. If that's a concern for a particular method we can leave it as an override here.